misc/xml-to-record-converter/src/main/java/io/ballerina/xmltorecordconverter/util/ConverterUtils.java

```diff
@@ -118,24 +117,24 @@ public static List<TypeDescriptorNode> extractTypeDescriptorNodes(List<TypeDescr
      * @return {@link List<TypeDescriptorNode>} The sorted TypeDescriptorNode list.
      */
     public static List<TypeDescriptorNode> sortTypeDescriptorNodes(List<TypeDescriptorNode> typeDescriptorNodes) {
-        List<TypeDescriptorNode> nonArrayNodes = typeDescriptorNodes.stream()
-                .filter(node -> !(node instanceof ArrayTypeDescriptorNode)).collect(Collectors.toList());
+        List<TypeDescriptorNode> nonArrayNodes = new ArrayList<>(typeDescriptorNodes.stream()
```

Comment:
let's leave this and `arrayNodes` as is so we can do the inplace sorting. then return a immutable list only at the end, using `toList`

Reply:
Fixed in [2778660](https://github.com/ballerina-platform/ballerina-lang/pull/42904/commits/277866060d74b08115089625bb7d582c095697a1)